File: src/SamuraiServer/SamuraiServer.Data/Models/GamePlayer.cs
using System;
using System.Collections.Generic;

namespace SamuraiServer.Data
{
    public class GamePlayer
    {
        public GamePlayer()
        {
            Units = new List<Unit>();
        }

        public Guid Id { get; set; }

        public Player Player { get; set; }

        public IList<Unit> Units { get; set; }

        public int Score { get; set; }

        public bool IsAlive { get; set; }
    }
}

File: src/Samurai.Client.JS/models/gamePlayer.js
define(function () {
  function GamePlayer(opt) {
    this.initialize(opt);
  }

  GamePlayer.prototype.initialize = function (opt) {
    var key = null,
        default_args = {
          //Guid
          Id  : '',
          //Player
          Player  : null,
          //IList<Unit>
          Units  : [],
          Score : 0,
          IsAlive : true,
        };

    opt || (opt = default_args);

    for(key in default_args) {
      if(typeof opt[key] == "undefined") opt[key] = default_args[key];
    }

    // opt[] has all the data - user provided and optional.
    for(key in opt) {
      this[key] = opt[key];
    }
  };

  return GamePlayer;
});